src/cmd_vel_mux.cpp

```diff
@@ -79,144 +82,253 @@ CmdVelMux::CmdVelMux(rclcpp::NodeOptions options) : rclcpp::Node("cmd_vel_mux",
+{
+  if (parameter_value.get_type() == rclcpp::ParameterType::PARAMETER_NOT_SET)
+  {
+    if (parsed_parameters.count(input_name) > 0)
     {
-      longest_timeout = list_[i]->timeout_;
+      parsed_parameters.erase(input_name);
     }
+    return true;
+  }
+  if (parsed_parameters.count(input_name) == 0)
+  {
+    parsed_parameters.emplace(std::make_pair(input_name, ParameterValues()));
   }
 
-  if (!common_timer_ || longest_timeout != (common_timer_period_ / 2.0))
+  if (parameter_name == "topic")
+  {
+    if (parameter_value.get_type() != rclcpp::ParameterType::PARAMETER_STRING)
+    {
+      RCLCPP_WARN(get_logger(), "topic must be a string; ignoring");
+      return false;
+    }
+    parsed_parameters[input_name].topic = parameter_value.as_string();
+  }
+  else if (parameter_name == "timeout")
   {
-    // Create another timer for cmd_vel messages from any source, so we can
-    // dislodge last active source if it gets stuck without further messages
-    common_timer_period_ = longest_timeout * 2.0;
-    common_timer_ = this->create_wall_timer(std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::duration<double>(common_timer_period_)), std::bind(&CmdVelMux::commonTimerCallback, this));
+    if (parameter_value.get_type() != rclcpp::ParameterType::PARAMETER_DOUBLE)
+    {
+      RCLCPP_WARN(get_logger(), "timeout must be a double; ignoring");
+      return false;
+    }
+    parsed_parameters[input_name].timeout = parameter_value.as_double();
   }
+  else if (parameter_name == "priority")
+  {
+    if (parameter_value.get_type() != rclcpp::ParameterType::PARAMETER_INTEGER)
+    {
+      RCLCPP_WARN(get_logger(), "priority must be an integer; ignoring");
+      return false;
+    }
 
-  RCLCPP_INFO(get_logger(), "CmdVelMux : (re)configured");
+    int64_t priority = parameter_value.as_int();
+    if (priority < 0 || priority > std::numeric_limits<uint32_t>::max())
+    {
+      RCLCPP_WARN(get_logger(), "Priority out of range, must be between 0 and MAX_UINT32");
+      return false;
+    }
+    if (used_priorities_.count(priority) != 0)
+    {
+      RCLCPP_WARN(get_logger(), "Cannot have duplicate priorities, ignoring");
+      return false;
+    }
+    used_priorities_.insert(priority);
+    parsed_parameters[input_name].priority = priority;
+  }
+  else if (parameter_name == "short_desc")
+  {
+    if (parameter_value.get_type() != rclcpp::ParameterType::PARAMETER_STRING)
+    {
+      RCLCPP_WARN(get_logger(), "short_desc must be a string; ignoring");
+      return false;
+    }
+    parsed_parameters[input_name].short_desc = parameter_value.as_string();
+  }
+  else
+  {
+    RCLCPP_WARN(get_logger(), "Invalid input variable '%s'; ignored", parameter_name.c_str());
+    return false;
+  }
+
+  return true;
+}
+
+std::map<std::string, ParameterValues> CmdVelMux::parseFromParametersMap(const std::map<std::string, rclcpp::Parameter> & parameters)
+{
+  used_priorities_.clear();
```

Comment:
@clalancette @Lobotuerk meta: I still think `used_priorities_` should not be a member field as it's tied to the `std::map` that this function populates and returns, but I don't mind merging as-is for the sake of moving forward.

Reply:
Oh yeah, good point.  In point of fact, we have all of the infrastructure in place to just do this checking inside of `parametersAreValid`, so I moved it into there in 0ec9740c7fb63875e82d5af4228b8373e10c8259